### DISCUSSION TITLE
504 Gateway timeout on Save

### DISCUSSION THREAD

**Discussion by phillipzada** [requarks/wiki#5691] [Error / Bug Report]
Hi,

We are getting a save a 504 Gateway timeout. When we refresh the page, the content seemed to have been saved correctly. We aren't seeing any errors in the log and just wondering if there was a way to increase the 30-second timeout to identify if it's just a resource problem.

**Comment by woehrl** [comment]
Same here! After restarting the wiki.js service, the problem was gone (for now). So it seems like this is something stacking up over longer runtime. Is there any information available to share, any logfiles? Can't see any problem in syslog...
Nothing very special activated on our installation, latest 2.5.289, postgresql, local auth. The only "new" thing: recently activated: an SFTP backup.

**Reply by theAkito** [reply]
I also use SFTP from the get-go. Using PostgreSQL, too. Local Authentication, as well.

**Comment by theAkito** [comment]
100% same issue here with proper Helm & Kubernetes setup.

A long while ago, I installed a self-compiled version of the product. I thought this was the culprit. Then I compiled a more proper version & everything started working again. I thought, it was just an error with the first version.

Now, after running this new & proper version for quite a while, the error re-appears! & it does not go away, without restarting.

This is a huge & important issue. Quite baffles me, how it isn't yet converted into an issue, yet.

**Comment by YunusEmreAlps** [comment]
We've successfully resolved the issue by extending the router timeout period on the OpenShift side. Annotations have been added to the router definition with the following configuration:

```yaml
annotations:
    haproxy.router.openshift.io/timeout: 3600s
```
This adjustment allows for a longer timeout period (1 hour), providing sufficient time for requests to be processed without encountering timeout errors.
